src/useResizeDetector.ts

```diff
@@ -42,18 +42,38 @@ function useResizeDetector<T extends HTMLElement = any>({
     }, 0);
   }
 
-  // this is a callback that will be called every time the ref is changed
+  // this is a memo that will be called every time the ref is changed
+  // This proxy will properly call setState either when the ref is called as a function or when `.current` is set
   // we call setState inside to trigger rerender
-  const onRefChange: OnRefChangeType = useCallback(
-    (node: T | null) => {
-      if (node !== refElement) {
-        setRefElement(node);
-      }
-    },
+
+  const onRefChange: OnRefChangeType<T> = useMemo(
```

Comment:
You should update `onRefChange` property name to better represent it's purpose.
Consider renaming it to `refProxy` or `ref`.

Reply:
Good point, fixed